src/googmodule.ts

```diff
@@ -0,0 +1,388 @@
+          null {
+        const importedUrl = extractRequire(call);
+        if (!importedUrl) return null;
+        const imp = importPathToGoogNamespace(host, sf, importedUrl);
+        modulesManifest.addReferencedModule(sf.fileName, imp.namespace.text);
+        const ident: ts.Identifier|undefined = namespaceToModuleIdentifier.get(imp.namespace.text);
+        let initializer: ts.Expression;
+        if (!ident) {
+          namespaceToModuleIdentifier.set(imp.namespace.text, newIdent);
+          initializer = createGoogCall('require', imp.namespace);
+        } else {
+          initializer = ident;
+        }
+        const varDecl = ts.createVariableDeclaration(newIdent, /* type */ undefined, initializer);
+        const newStmt = ts.createVariableStatement(
+            /* modifiers */ undefined, ts.createVariableDeclarationList([varDecl]));
+        return ts.setOriginalNode(ts.setTextRange(newStmt, original), original);
+      }
+
+      function visitTopLevelStatement(
+          statements: ts.Statement[], sf: ts.SourceFile, node: ts.Statement): void {
+        switch (node.kind) {
+          case ts.SyntaxKind.ExpressionStatement: {
+            const exprStmt = node as ts.ExpressionStatement;
+            // Check for "use strict" and certain Object.defineProperty and skip it if necessary.
+            if (isUseStrict(exprStmt) || isEsModuleProperty(exprStmt)) {
+              stmts.push(ts.createNotEmittedStatement(exprStmt));
+              return;
+            }
+            // Check for:
+            //   module.exports = ...;
+            const modExports = rewriteModuleExportsAssignment(exprStmt);
+            if (modExports) {
+              stmts.push(modExports);
+              return;
+            }
+            // Check for:
+            //   "require('foo');" (a require for its side effects)
+            const expr = exprStmt.expression;
+            if (!ts.isCallExpression(expr)) break;
+            let callExpr = expr;
+            // Handle export * in ES5 mode (in ES6 mode, export * is dereferenced already).
+            // export * creates either a pure top-level '__exportStar(require(...))' or the imported
+            // version, 'tslib.__exportStar(require(...))'. The imported version is only substituted
+            // later on though, so appears as a plain "__exportStar" on the top level here.
+            const isExportStar =
+                (ts.isIdentifier(expr.expression) && expr.expression.text === '__exportStar') ||
+                (ts.isIdentifier(expr.expression) && expr.expression.text === '__export');
+            if (isExportStar) callExpr = expr.arguments[0] as ts.CallExpression;
+            const ident = ts.createIdentifier(nextModuleVar());
+            const require = maybeCreateGoogRequire(exprStmt, callExpr, ident);
+            if (!require) break;
+            statements.push(require);
+            // If this is an export star
```

Comment:
Seems redundant with the following line, maybe commit got snipped?

Reply:
Completed the comment.